### PR TITLE
[REVIEW] Fix `build.sh` when `PARALLEL_LEVEL` env variable isn't set

### PR DESCRIPTION
**Pull request by kkraus14**
Currently if `PARALLEL_LEVEL` is not set, it defaults to an empty string. Later in the script this causes a command to look like: `python setup.py build_ext --inplace -j` which isn't allowed as for Python you need to add a number of threads after `-j`.

To fix this, defaulting `PARALLEL_LEVEL` to the output of `nproc` which should be the default behavior of passing `-j` anyway.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: build.sh [hunk 1]
```diff
@@ -65,7 +65,7 @@ BUILD_PER_THREAD_DEFAULT_STREAM=OFF
 #  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
 #         CONDA_PREFIX, but there is no fallback from there!
 INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}
-PARALLEL_LEVEL=${PARALLEL_LEVEL:=""}
+PARALLEL_LEVEL=${PARALLEL_LEVEL:=$(nproc)}
```
Comment (dillon-cullinan) [inline build.sh:68]:
```suggestion
PARALLEL_LEVEL=${PARALLEL_LEVEL:=4}
```

**Review by dillon-cullinan — CHANGES_REQUESTED** [review]
`nproc` will use all available CPU cores. This isn't viable for gpuCI which does multiple builds on single machines.

**Review by dillon-cullinan — APPROVED** [review]
gpuCI should define PARALLEL_LEVEL manually in build scripts. So this should be fine.

**Pull request merged**